jax/lax/lax.py

```diff
@@ -84,17 +84,7 @@ def _canonicalize_shape(shape):
   # TODO(mattjj): this next check is a temporary workaround for masking
   if (type(shape) is tuple and masking.is_polymorphic(shape)):
     return shape
```

Comment:
This case is actually handled inside `abstract_arrays.canonicalize_shape` already, so we could probably safely drop these lines, too.

Reply:
I can believe it's not necessary anymore, but I don't follow where it's handled in `abstract_arrays.canonicalize_shape`. `is_polymorphic` is checking if shape elements are subclasses of Poly, and there's no such code in `abstract_arrays`.

Reply:
It's hidden inside `_canonicalize_dimension`